web-client/src/presenter/computeds/trialLocationHelper.ts

```diff
@@ -0,0 +1,135 @@
+import { formatDateString } from '@shared/business/utilities/DateHandler';
+import { state } from '@web-client/presenter/app.cerebral';
+import { BlockedFormattedCase } from '@web-client/presenter/computeds/blockedCasesReportHelper';
+
+export const trialLocationHelper = (
+  get: Get,
+): {
+  location: string;
+  eligibleCasesForDisplay: any[];
+  formattedBlockedCases: any[];
+  formattedEligibleCases: any[];
+  totalPagesEligible: number;
+  blockedCasesForDisplay: any[];
+  totalPagesBlocked: number;
+} => {
+  const pageSize = 100;
+
+  const { blockedCasesPage, eligibleCases, eligibleCasesPage, location } = get(
+    state.trialLocationPage,
+  );
+
+  const blockedCases = get(state.blockedCases);
+
+  const formattedBlockedCases = blockedCases
+    .sort(compareCasesByDocketNumber)
+    .map(blockedCase =>
+      setConsolidationFlagsForDisplay(blockedCase, blockedCases),
+    )
+    .map(blockedCase => {
+      const updatedCase = {
+        ...setFormattedBlockDates(blockedCase),
+        caseTitle: Case.getCaseTitle(blockedCase.caseCaption || ''),
+      };
+
+      return updatedCase;
+    });
+
+  const formattedEligibleCases = eligibleCases.map(c => {
+    const isDocketSuffixHighPriority = HIGH_PRIORITY_SUFFIXES.includes(
+      c.docketNumberSuffix!,
+    );
+    const caseTitle = Case.getCaseTitle(c.caseCaption);
+
+    return {
+      ...c,
+      caseTitle,
+      isDocketSuffixHighPriority,
+    };
+  });
+
+  const groups = getPriorityGroups(formattedEligibleCases);
+
+  const sortedEligibleCases = formattedEligibleCases
+    .map(caseItem => {
+      return addGroupSymbol(
+        setConsolidationFlagsForDisplay(
+          caseItem,
+          groups[caseItem[groupKeySymbol]],
+        ),
+        caseItem[groupKeySymbol],
+      );
+    })
+    .sort(compareTrialSessionEligibleCases(formattedEligibleCases));
+
+  const eligibleCasesForDisplay = sortedEligibleCases.slice(
+    eligibleCasesPage * pageSize,
+    eligibleCasesPage * pageSize + pageSize,
+  );
+
+  const blockedCasesForDisplay = formattedBlockedCases.slice(
+    blockedCasesPage * pageSize,
+    blockedCasesPage * pageSize + pageSize,
+  );
+
+  return {
+    blockedCasesForDisplay,
+    eligibleCasesForDisplay,
+    formattedBlockedCases,
+    formattedEligibleCases,
+    location,
+    totalPagesBlocked: Math.ceil(formattedBlockedCases.length / pageSize),
+    totalPagesEligible: Math.ceil(sortedEligibleCases.length / pageSize),
+  };
+};
+
+const setFormattedBlockDates = (
+  blockedCase: RawCase & {
+    inConsolidatedGroup: boolean;
+    consolidatedIconTooltipText: string;
+    shouldIndent: boolean;
+    isLeadCase: boolean;
+  },
+): BlockedFormattedCase => {
+  const blockedFormattedCase: BlockedFormattedCase = {
+    ...blockedCase,
+    blockedDateEarliest: '',
+    caseTitle: '',
+  };
+
+  if (blockedCase.blockedDate && blockedCase.automaticBlocked) {
```

Comment:
It looks like the top-level conditions are all based on `blocked` or `automaticBlocked` except this first `blockedCase.blockedDate`. Is that correct? (Just worried about any possible discrepancy between blocked and blockedDate in this logic.)

Reply:
This _is_ confusing. Refactored to use the same method between `blockedCasesReportHelper` and `trialLocationHelper`